pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -900,3 +950,158 @@ func (r *ReconcileBareMetalHost) publishEvent(request reconcile.Request, event c
 func hostHasFinalizer(host *metal3v1alpha1.BareMetalHost) bool {
 	return utils.StringInList(host.Finalizers, metal3v1alpha1.BareMetalHostFinalizer)
 }
+
+func(r *ReconcileBareMetalHost) buildAndValidateCleanSteps(
+	request reconcile.Request, host *metal3v1alpha1.BareMetalHost, prov provisioner.Provisioner) (cleanSteps []nodes.CleanStep) {
+	reqLogger := log.WithValues(
+		"Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	// Retrieve RAID configuration in form of provisioner's clean step
+	raidCleanStep, err := r.ValidateAndBuildRAIDCleanStep(&host.Spec.RAID)
+	if err != nil {
+		reqLogger.Info("Failed to retrieve RAID configuration", err)
+	} else if raidCleanStep != nil {
+		// ‘create_configuration’ doesn’t remove existing disks. It is recommended
+		// to add ‘delete_configuration’ before ‘create_configuration’ to make sure
+		// that only the desired logical disks exist in the system after manual cleaning.
+		cleanSteps = append(
+			cleanSteps,
+			nodes.CleanStep{
+				Interface: "raid",
+				Step:      "delete_configuration",
+				Args:      nil,
+			},
+			*raidCleanStep,
+		)
```

Comment:
For software RAID we'll also need to wipe partitions. But this is for Train.

Reply:
Thanks @dtantsur for Software RAID case. I will take care it too :)